plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2764,7 +2764,8 @@ public void validateScan(ConnectorSession session, ConnectorTableHandle handle)
                 }
                 Set<String> referencedColumns =
                         deltaLakeTableHandle.getConstraintColumns().stream()
-                                .map(DeltaLakeColumnHandle::getBaseColumnName)
+                                // Lowercase column names because the above partitionColumns variable uses canonical names
+                                .map(column -> column.getBaseColumnName().toLowerCase(ENGLISH))
```

Comment:
"above partitionColumns variable uses canonical names" 

i forgot that "canonical" stands for "lowercase".
it's very unfortunate.
let's rename `io.trino.plugin.deltalake.transactionlog.MetadataEntry#getCanonicalPartitionColumns` to `getLowercasePartitionColumns`

here, we should remove toLowerCase and switch to `getOriginalPartitionColumns`